elastic/apm-agent-nodejs
# refactor: drop unused 'npm run test:cli'; add test for apm.middleware.connect()

Let's drop test/script/cli.js. It is a little CLI, exposed as `npm run test:cli`,
that prompts interactively for which of these test types to run:

```js
    case 'standard': return standardTest()
    case 'http': return httpTest()
    case 'restify': return restifyTest()
    case 'connect': return connectTest()
    case 'express': return expressTest()
    case 'transaction': return transactionTest()
```

The select test is run, which generates some APM data (sent to the default
`serverUrl`).

- standard: calls `apm.captureError()` a few times
- http: exercises `http.createServer()` and `http.get()` instrumentation
- restify: exercises `restify.createServer()` and restify-clients' `client.get()`
- connect: exercises `connect` instrumentation
- express: exercises `express` instrumentation
- transaction: spends 65s generating transactions with some child spans, e.g.:
    ```
    transaction dc3e29 "foo"
    `- span fa63c8 "sig1"
      `- span 8da1e6 "sig2"
        `- span 46866e "sig5"
          `- span e4d3b6 "sig6"
        `- span 36309d "sig3"
          `- span b5f043 "sig4"
    ```

This appears to be the only place that `connect` and `restify-clients` are
exercised under "test/...". This exercising useless because no one runs this
(correct me if I'm wrong there) and the output isn't checked.

- connect: I wonder if this module is pretty much EOL already. The last release
  was `"3.7.0": "2019-05-18T00:58:45.162Z"`. The repo is getting what looks like
  automated commits. **We can add a small test/connect.test.js for this.** Adding
  TAV tests is overkill.
  NOTE: Adding this test turned up an issue: https://github.com/elastic/apm-agent-nodejs/issues/2437
- restify-clients: **I see no value in adding more test code for restify-clients.**
  We don't document support for restify-clients, we have never otherwise
  tested it. The restify-clients just use the Node.js core `http.request`
  under the hood.


### Checklist

- [x] Implement code
